### ISSUE TITLE
get_bams.R error

### ISSUE THREAD

**Issue by Jwong684** [LCR-BCCRC/lcr-scripts#3]
I get this error with the get_bams.R:

```
Setting up session...
Retrieving list of merged BAM files...
Warning message:
The following library IDs do not exist:
Warning messages:
1: `tz` argument is ignored by `as_date()`
2: `tz` argument is ignored by `as_date()`
Warning message:
Unknown library strategy encountered: EXC-Seq, NA
Warning message:
These libraries do not have merges: B22792, B22861, B22793, B22862, B22784, B22853, B22795, B22864, B22798, B22867, B22803, B22872, B59260, B59255, B59261, B59238, B59235, B59236, B59251, B59237, B59243, B59256, B59244, B59245, B59240, B59259, B59252, B59254, B59241, B59249, B59247, B59248, B59253, B59264, B59246, B59263, B59262, B59267, B59268, B59269, B59270, B59239, B58144, B58145, B59266, B59265, B59258, B58146, B58129, B58130, B58131, B58132, B58133, B58134, B58135, B58136, B58137, B58138, B58139, B58140, B58141, B58142, B58143, B59242, B59250, B59257, B59267, B59268, B59269, B59270
Retrieving RNA-seq aligned_libcore IDs...
Retrieving aligned_libcore Bio QC status and comments...
Error: Problem with `summarise()` input `lc.comments_warning`.
x 'x' must be atomic
i Input `lc.comments_warning` is `(structure(function (..., .x = ..1, .y = ..2, . = ..1) ...`.
i The error occured in group 1: bam_id = 8.
Backtrace:
     x
  1. \-`%>%`(...)
  2.   +-base::withVisible(eval(quote(`_fseq`(`_lhs`)), env, env))
  3.   \-base::eval(quote(`_fseq`(`_lhs`)), env, env)
  4.     \-base::eval(quote(`_fseq`(`_lhs`)), env, env)
  5.       \-`_fseq`(`_lhs`)
  6.         \-magrittr::freduce(value, `_function_list`)
  7.           +-base::withVisible(function_list[[k]](value))
  8.           \-function_list[[k]](value)
  9.             \-dplyr::summarise_at(...)
 10.               +-dplyr::summarise(.tbl, !!!funs)
 11.               \-dplyr:::summarise.grouped_df(.tbl, !!!funs)
 12.                 \-dplyr:::summarise_cols(.da
In addition: Warning message:
Values are not uniquely identified; output will contain list-cols.
* Use `values_fn = list` to suppress this warning.
* Use `values_fn = length` to identify where the duplicates arise
* Use `values_fn = {summary_fun}` to summarise duplicates
Execution halted
```

I found out that the bug comes from the command to split comments by status and collapse using libcore IDs (~line 364). More specifically, the error comes out at summarise_at(), but the warning messages stem from pivot_wider().

To fix:
Group by status, then create a dummy variable for each "status" name before using pivot_wider()

new code:
  ```
bio_qc_comments <-
    bio_qc_info %>%
    dplyr::filter(!is.na(bio_qc_comments), bio_qc_comments != "") %>%
    separate_rows(bio_qc_comments, sep = "; ?(?=Failed|Warning|Manual)") %>%
    mutate(bio_qc_comments = ifelse(grepl(":", bio_qc_comments),
                                    bio_qc_comments,
                                    paste0("Other:", bio_qc_comments))) %>%
    separate(bio_qc_comments, c("status", "comment"),
             sep = ":", extra = "merge") %>%
    mutate(status = sub(" ", "_", status),
           status = tolower(status),
           status = paste0("lc.comments_", status),
           comment = sub('"', "", comment),
           comment = trimws(comment),
           comment = paste0(aln_libcore_id, "={", comment, "}")) %>% 
    group_by(status) %>%
    mutate(row = row_number()) %>%
    pivot_wider(names_from = status, values_from = comment) %>% 
    dplyr::select(-row) %>% 
    group_by(bam_id) %>% 
    summarise_at(vars(contains("comments_")),
                 ~ paste(sort(na.omit(unique(.))), collapse = "|"))
```

**Comment by BrunoGrandePhD** [comment]
@Jwong684: Could you post the command you ran? I want to check why `pivot_wider()` isn't resulting in unique values.

**Comment by Jwong684** [comment]
I used this:
``` 
/projects/dscott_prj/CCSRI_1500/lowpassWGS/envs/ichorcna/bin/Rscript
/projects/bgrande/experiments/2020-01-29-get_bams_demo/get_bams.R
/projects/dscott_prj/CCSRI_1500/exomes/library_info.txt
/projects/dscott_prj/CCSRI_1500/exomes/data/metadata/tmp.tsv Library 
```

**Comment by lkhilton** [comment]
It would, definitely. I can make a branch of the repo and put my version there. 

I think the warning message from `pivot_wider` is expected and supposed to be handled in subsequent steps.

**Comment by BrunoGrandePhD** [comment]
Great! You can create a branch and open a PR. I'll review the error and see how to prevent it and contribute to the PR. Thanks, @lkhilton!

**Comment by lkhilton** [comment]
@Jwong684 was there a particular library ID associated with `bam_id=8`? This is probably an edge case from the database and it would be helpful on review to know what exactly the API is returning.

**Comment by Jwong684** [comment]
bam_id = 8 (taken from the bam_files output) is library: B22801. There is only one library associated with bam_id = 8.

From the command that shot the warning, there doesn't seem to be anything off with bam_id = 8; it is just the first sample in the table.

**Comment by Jwong684** [comment]
From the code that shot the error:
Table output before error:
```    
bio_qc_info %>%
    dplyr::filter(!is.na(bio_qc_comments), bio_qc_comments != "") %>%
    separate_rows(bio_qc_comments, sep = "; ?(?=Failed|Warning|Manual)") %>%
    mutate(bio_qc_comments = ifelse(grepl(":", bio_qc_comments),
                                    bio_qc_comments,
                                    paste0("Other:", bio_qc_comments))) %>%
    separate(bio_qc_comments, c("status", "comment"),
             sep = ":", extra = "merge") %>%
    mutate(status = sub(" ", "_", status),
           status = tolower(status),
           status = paste0("lc.comments_", status),
           comment = sub('"', "", comment),
           comment = trimws(comment),
           comment = paste0(aln_libcore_id, "={", comment, "}")) %>% View() 
```

![image](https://user-images.githubusercontent.com/22081472/84426070-7952a880-abd7-11ea-8797-50efaad35286.png)



Table output that gives the warning (after including pivot_wider()):
```
bio_qc_info %>%
    dplyr::filter(!is.na(bio_qc_comments), bio_qc_comments != "") %>%
    separate_rows(bio_qc_comments, sep = "; ?(?=Failed|Warning|Manual)") %>%
    mutate(bio_qc_comments = ifelse(grepl(":", bio_qc_comments),
                                    bio_qc_comments,
                                    paste0("Other:", bio_qc_comments))) %>%
    separate(bio_qc_comments, c("status", "comment"),
             sep = ":", extra = "merge") %>%
    mutate(status = sub(" ", "_", status),
           status = tolower(status),
           status = paste0("lc.comments_", status),
           comment = sub('"', "", comment),
           comment = trimws(comment),
           comment = paste0(aln_libcore_id, "={", comment, "}")) %>% pivot_wider(names_from = status, values_from = comment) %>% View()
```

![image](https://user-images.githubusercontent.com/22081472/84426177-a30bcf80-abd7-11ea-8573-4071c6991874.png)

This output will then trigger the error after being piped into summarise_at(sort()):
```
Error in sort.int(x, na.last = na.last, decreasing = decreasing, ...) : 
  'x' must be atomic
```

**Comment by BrunoGrandePhD** [comment]
Addressed in #5.
